### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by fpsapc**
- Write unit tests with React Testing Library.
- Write unit tests with snapshots with Jest.
- Write unit tests for the files operate.js and calculate.js using Jest
- Create unit tests for all React components
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.
- Run full tests suite using Jest. All tests passed.

### CODE REVIEW

File: src/components/tests/Calculator.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Tufahel) [inline src/components/tests/Calculator.test.js]: - [x] Kindly avoid using comments to skip linter errors, as it is not a good practice. Sometimes linter error happens due to syntax error or anything related to the code. So try to fix it manually and this way you may learn a new concept.

File: src/components/tests/Home.test.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Tufahel) [inline src/components/tests/Home.test.js]: - [x] Kindly avoid using comments to skip linter errors, as it is not a good practice. Sometimes linter error happens due to syntax error or anything related to the code. So try to fix it manually and this way you may learn a new concept.

File: src/components/tests/Layout.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Tufahel) [inline src/components/tests/Layout.test.js]: - [x] Kindly avoid using comments to skip linter errors, as it is not a good practice. Sometimes linter error happens due to syntax error or anything related to the code. So try to fix it manually and this way you may learn a new concept.

File: src/components/tests/Quote.test.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Tufahel) [inline src/components/tests/Quote.test.js]: - [x] Kindly avoid using comments to skip linter errors, as it is not a good practice. Sometimes linter error happens due to syntax error or anything related to the code. So try to fix it manually and this way you may learn a new concept.

**Review by Tufahel — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @fpsapc & @FevenSeyfu :wave:  

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🌟

Alt Text

✔️ Professional pull request
✔️ Correct git-flow
✔️ Linter checks are passing

## Required Changes ♻️

***Please kindly check the comments below.***

## Optional suggestions 🤷‍♂️

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/tests/Calculator.test.js (hunk 1 above)
Comment (fpsapc) [reply src/components/tests/Calculator.test.js]: @Tufahel : Thanks for your kind review please help me to fix it. I found this error on that line
'react-test-renderer' should be listed in the project's dependencies. Run 'npm i -S react-test-renderer' to add iteslint[import/no-extraneous-dependencies](https://github.com/import-js/eslint-plugin-import/blob/v2.27.5/docs/rules/no-extraneous-dependencies.md) 
and I already have react-test-renderer installed and in the dependencies list. You can also see it in the screenshot 
fix

**Review by DuaneDave — APPROVED** [review]
Hi @fpsapc 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ All test are running well

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**